app/src/main/java/com/github/se/orator/ui/friends/FriendsScreen.kt

```diff
@@ -362,6 +378,39 @@ fun ViewFriendsScreen(
                     }
                   }
 
+              // Display the BattlePopup if a battle is selected
+              selectedBattle?.let { battle ->
+                // Extract the InterviewContext from the SpeechBattle
+                val battleContext = battle.context
+                // Get the challenger's name
+                val challengerName = userProfileViewModel.getName(battle.challenger)
+                BattlePopup(
+                    challengerName = challengerName,
+                    interviewContext = battleContext,
+                    onAccept = {
+
+                      // Handle battle acceptance
+                      battleViewModel?.acceptBattle(battle.battleId)
+                      selectedBattle = null
+
+                      // Show a toast
+                      Toast.makeText(localContext, "Battle accepted!", Toast.LENGTH_SHORT).show()
+                    },
+                    onDecline = {
+
+                      // Handle battle decline
+                      battleViewModel?.declineBattle(battle.battleId)
+                      selectedBattle = null
+
+                      // Show a toast
+                      Toast.makeText(localContext, "Battle accepted!", Toast.LENGTH_SHORT).show()
```

Comment:
I think the text should be "Battle declined.".

```suggestion
                      Toast.makeText(localContext, "Battle declined.", Toast.LENGTH_SHORT).show()
```